pdns/dnsdist-web.cc

```diff
@@ -58,68 +72,77 @@ static void connectionThread(int sock, ComboAddress remote, string password)
 
     string command=req.getvars["command"];
 
-    string callback;
-
-    if(req.getvars.count("callback")) {
-      callback=req.getvars["callback"];
-      req.getvars.erase("callback");
-    }
-
     req.getvars.erase("_"); // jQuery cache buster
 
     YaHTTP::Response resp(req);
+    resp.headers["X-Content-Type-Options"] = "nosniff";
+    resp.headers["X-Frame-Options"] = "deny";
+    resp.headers["X-Permitted-Cross-Domain-Policies"] = "none";
+    resp.headers["X-XSS-Protection"] = "1; mode=block";
+    resp.headers["Content-Security-Policy"] = "default-src 'self'; img-src *; style-src 'self' 'unsafe-inline'";
 
```

Comment:
Because the current index.html loads images from s3.amazonaws.com and www.powerdns.com, and allowing s3.amazonaws.com felt like allowing pretty much anything. Do you think it should be more restrictive?